test/contex_point_plot_test.exs

```diff
@@ -0,0 +1,182 @@
+      assert plot.colour_palette == :default
+    end
+  end
+
+  describe "set_size/3" do
+    test "updates the height and width", %{plot: plot} do
+      plot = PointPlot.set_size(plot, 666, 222)
+      assert plot.width == 666
+      assert plot.height == 222
+    end
+  end
+
+  describe "get_svg_legend/1" do
+
+    test "returns properly formatted legend", %{plot: plot} do
+      {:safe, svg} =
+        Contex.Plot.new(150, 150, plot)
+        |> Contex.Plot.plot_options(%{legend_setting: :legend_right})
+        |> Contex.Plot.to_svg()
+      
+      legend =
+        IO.chardata_to_string(svg) 
+        |> xpath(~x"//g[@class='exc-legend']", 
+             box: [
+               ~x"./rect", 
+               x: ~x"./@x"s,
+               y: ~x"./@y"s,
+               height: ~x"./@height"s,
+               width: ~x"./@width"s,
+               style: ~x"./@style"s
+             ],
+             text: [
+               ~x"./text", 
+               x: ~x"./@x"s,
+               y: ~x"./@y"s,
+               text_anchor: ~x"./@text-anchor"s,
+               dominant_baseline: ~x"./@dominant-baseline"s,
+               text: ~x"./text()"s
+             ]
+           )
+
+      # The other attributes are not tested because they are hard-coded.
+      assert %{y: "0", style: "fill: #1f77b4;"} = legend.box
+      assert %{y: "9", text: "bb"} = legend.text
+    end
+  end
+
+  describe "to_svg/1" do
+
+    defp plot_iodata_to_map(plot_iodata) do
+      IO.chardata_to_string(plot_iodata)
+      |> String.replace_prefix("", "<svg>")
+      |> String.replace_suffix("", "</svg>")
+      |> xpath(~x"//g/circle"l, [
+        cx: ~x"./@cx"s,
+        cy: ~x"./@cy"s,
+        style: ~x"./@style"s
+      ])
+    end
+
+    # Axis svg not tested as it is for practical purposes handled 
+    # by Contex.Axis, which is tested by ContexAxisTest
+    test "returns properly constructed chart", %{plot: plot} do
+      points_map =
+        PointPlot.to_svg(plot)
+        |> plot_iodata_to_map()
+
+      assert ["fill: #1f77b4;"] ==
+        Enum.map(points_map, fn point -> Map.get(point, :style) end)
+        |> Enum.uniq()
+
+      assert [{57.143, 42.857}, {100, 0}, {0, 100}] ==
+        Enum.map(points_map, fn point -> 
+          {Map.get(point, :cx)
+           |> String.to_float()
+           |> Float.round(3),
+           Map.get(point, :cy)
+           |> String.to_float()
+           |> Float.round(3)
+          } 
+        end)
+    end
+
+    test "renders custom fill colors properly", %{plot: plot} do
+      points_map =
+        PointPlot.set_colour_col_name(plot, "d")
+        |> PointPlot.to_svg()
+        |> plot_iodata_to_map()
+
+      assert 2 ==
+        Enum.map(points_map, fn point -> Map.get(point, :style) end)
+        |> Enum.uniq()
+        |> Enum.count()
+
+    end
+  end
+
+  # TODO
+  # Not sure what happens if column name is invalid
+  # Seems like you should also be allowed to specify an index
+  # Need to test reset of scale
```

Comment:
With reworking the interface I think it makes sense to defer scale calcs until to_svg is called rather than each time a column or plot size is updated.